src/components/burger-constructor/burger-constructor.jsx

```diff
@@ -1,36 +1,61 @@
-import { useState } from "react";
-import PropTypes from "prop-types";
+import { useState, useContext, useEffect, useReducer } from "react";
 import styles from "./burger-constructor.module.css";
 import {
   ConstructorElement,
   DragIcon,
   Button,
 } from "@ya.praktikum/react-developer-burger-ui-components";
-import { dataTemplate } from "../../utils/constants";
 import Modal from "../modal/modal";
 import OrderDetails from "../order-details/order-details";
+import { PickedIngredients } from "../services/burgerContext";
 
-const BurgerConstructor = ({ data }) => {
-  const bun = data.filter((element) => element.type === "bun")[0];
+const refillConstructor = (state, action) => {
+  switch (action.type) {
+    case 'add':
+      return [...state, action.payload]
+    case 'delete':
+      return state.filter((el) => el._id !== action.payload._id)
+    default:
+      throw new Error(`Wrong type of action: ${action.type}`);
+  }
+}
+
+const BurgerConstructor = () => {
+  const {data, bun} = useContext(PickedIngredients);
   const [modal, setModal] = useState(false);
+  const [ingredients, dispatchIngredients] = useReducer(refillConstructor, []);
+  const [finalPrice, setFinalPrice] = useState(0);
+  const [ingredientsId, setIngredientsId] = useState()
+
+  useEffect(() => {
+    data.forEach((el) => {
+      dispatchIngredients({type : 'add', payload: el});
+    })
+  }, [data])
+
+  useEffect(() => {
+    setFinalPrice(ingredients.reduce((prev, curr) => prev + curr.price, 0) + (bun.price * 2));
```

Comment:
Тут можно еще хитрее сделать)
```suggestion
    setFinalPrice(ingredients.reduce((prev, { price }) => prev + price, bun.price * 2));
```

Reply:
Спасибо! Сам бы не догадался)